File: framework/extracted/Ahorn/entities/AshPhoenix.jl
module AshPhoenix

using ..Ahorn, Maple

@mapdef Entity "MaggyHelper/AshPhoenix" AshPhoenix(x::Integer, y::Integer, detectionRange::Integer=250, flySpeed::Integer=100, health::Integer=3)

const placements = Ahorn.PlacementDict(
    "normal" => Ahorn.EntityPlacement(AshPhoenix)
)

function Ahorn.selection(entity::AshPhoenix)
    return Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::AshPhoenix, room::Maple.Room)
    Ahorn.drawRectangle(ctx, Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16), (0.5, 0.5, 0.5, 0.5), (0.0, 0.0, 0.0, 1.0))
end

end

File: framework/extracted/Loenn/entities/ashPhoenix.lua
local utils = require("utils")
local drawableSprite = require("structs.drawable_sprite")

local ashPhoenix = {}

ashPhoenix.name = "MaggyHelper/AshPhoenix"
ashPhoenix.depth = -100
ashPhoenix.fieldInformation = {
    health = {
        minimumValue = 1
    },
    flySpeed = {
        minimumValue = 0
    },
    detectionRange = {
        minimumValue = 0
    }
}
ashPhoenix.placements = {
    {
        name = "normal",
        data = {
            health = 3,
            flySpeed = 100,
            detectionRange = 250
        }
    }
}

function ashPhoenix.sprite(room, entity)
    local sprite = drawableSprite.fromTexture("characters/ash_phoenix/fly", entity)
    sprite:setPosition(entity.x, entity.y)
    sprite:setJustification(0.5, 0.5)
    sprite.color = {1.0, 0.5, 0.0, 1.0}
    return sprite
end

function ashPhoenix.selection(room, entity)
    return utils.rectangle(entity.x - 16, entity.y - 14, 32, 28)
end

return ashPhoenix
